src/com/ms/silverking/cloud/dht/daemon/storage/NamespaceStore.java

```diff
@@ -669,7 +669,7 @@ private VersionCheckResult checkPutVersion(DHTKey key, long version, long requir
                 segmentNewestVersion = newestVersion(key);
                 if (segmentNewestVersion >= 0) {
                     if (debugVersion) {
-                        System.out.printf("version %d segmentNewestVersion %d\n", version, segmentNewestVersion);
+                        Log.fineAsyncf("version %d segmentNewestVersion %d", version, segmentNewestVersion);
```

Comment:
debug is off by default. Make Log.warningAsyncf so that the debug flag has an effect